src/lib/plugin_apis/fs.api

```diff
@@ -409,6 +409,32 @@ gboolean bd_fs_clean (const gchar *device, GError **error);
  */
 gchar* bd_fs_get_fstype (const gchar *device,  GError **error);
 
+/**
+ * bd_fs_freeze:
+ * @device: the device to freeze (device must be mounted, trying to freeze an
+ *          unmounted device will result in an error)
+ * @error: (out): place to store error (if any)
+ *
+ * Freezes filesystem on @device. The filesystem must support freezing.
+ *
+ * Returns: whether @device was successfully freezed or not
+ *
+ */
+gboolean bd_fs_freeze (const gchar *device, GError **error);
```

Comment:
Any reason why this takes a device and not a mount point?

Reply:
Because all our functions take devices and not mountpoints. Even `bd_fs_xfs_info` takes device. I can change this to mountpoint (it definitely makes more sense) but I thought it might be better to keep the API consistent.

Reply:
IIRC, `bd_fs_xfs_info()` takes a device to be consistent with the other `fs_*_info()` functions. But that doesn't mean that we need to do it here too. Let's prefer simplicity here. :)

Reply:
Ok, I've changed the functions to take mountpoint. I've also added a new functions to check if path is a mountpoint.